pkg/dequeuer/async_handler.go

```diff
@@ -106,7 +106,16 @@ func (h *AsyncMessageHandler) handleMessage(requestID string) error {
 	}
 	defer h.deletePayload(requestID)
 
-	result, err := h.submitRequest(payload, requestID)
+	headers, err := h.getHeaders(requestID)
+	if err != nil {
+		updateStatusErr := h.updateStatus(requestID, async.StatusFailed)
+		if updateStatusErr != nil {
+			h.log.Errorw("failed to update status after failure to get headers", "id", requestID, "error", updateStatusErr)
+		}
+		return errors.Wrap(err, "failed to get payload")
+	}
+
+	result, err := h.submitRequest(payload, headers, requestID)
```

Comment:
Can't comment on `func (h *AsyncMessageHandler) getPayload(requestID string) (*userPayload, error)` line, so I'm adding this comment here. I think we should remove the `ContentType` field from the `userPayload` struct and not set it to a default value if we don't find it.

We shouldn't make assumptions about what the content type is if it's not set.

What do you say?

Reply:
It seems to me like `payload.ContentType` is not being used currently, so I think we can just delete it from the struct and it won't have an impact